Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -8399,12 +8399,15 @@ bool WebPage::isUsingUISideCompositing() const
 }
 
 #if ENABLE(NETWORK_CONNECTION_INTEGRITY)
-void WebPage::setLookalikeCharacterStrings(Vector<String>&& strings)
+void WebPage::setLookalikeCharacterStrings(Vector<WebCore::LookalikeCharactersSanitizationData>&& strings)
 {
     m_lookalikeCharacterStrings.clear();
     m_lookalikeCharacterStrings.reserveInitialCapacity(strings.size());
-    for (auto& string : strings)
-        m_lookalikeCharacterStrings.add(string);
+    for (auto& data : strings) {
+        if (!m_lookalikeCharacterStrings.contains(data.domain))
+            m_lookalikeCharacterStrings.add(data.domain, HashSet<String> { });
+        m_lookalikeCharacterStrings.get(data.domain).add(data.lookalikeCharacters);
```

Comment:
There's a bit of redundant hashing here — perhaps we could use `HashMap::ensure()` instead, passing in a lambda that returns a new HashSet and then call `add` on the new or extant set, using the `AddResult` iterator.